### PR TITLE
X11: fix size/position (test video_setWindowCenteredOnDisplay)

### PR DESCRIPTION
**Pull request by 1bsyl**
X11: fix size/position (test video_setWindowCenteredOnDisplay)
this fix x11 backend to correctly pass video_setWindowCenteredOnDisplay()
- get border values early  (eg status bar)
- add border values, to have correct position sets.
- wait for size/position change to get valid values

### CODE REVIEW

**Comment by 1bsyl** [comment]
with this, I got: `Assert Summary: Total=104 Passed=104 Failed=0`
( before `Assert Summary: Total=104 Passed=92 Failed=12 ` )

File: src/video/x11/SDL_x11window.c [hunk 1]
```diff
@@ -1516,6 +1526,17 @@ static void X11_SetWindowFullscreenViaWM(_THIS, SDL_Window *window, SDL_VideoDis
...
         }
 
+        if (!fullscreen) {
+            int dest_x = 0, dest_y = 0;
+            SDL_RelativeToGlobalForWindow(window,
+                                          window->windowed.x, window->windowed.y,
+                                          &dest_x, &dest_y);
+
+            /* Attempt to move the window */
+            X11_XMoveWindow(display, data->xwindow, dest_x, dest_y);
```
Comment (slouken) [inline src/video/x11/SDL_x11window.c:1536]: Are we missing the border values here?

Comment (1bsyl) [reply src/video/x11/SDL_x11window.c:1536]: no, I think this ok!
but there are more stuff to fix I think!

@slouken: just a question to make sure,
SDL_SetWindowPosition(x, y).  
x/y is the top-left of the window decoration border right ? (topleft rendering area, is little after eg (x, y+50))

Comment (slouken) [reply src/video/x11/SDL_x11window.c:1536]: x/y is the top-left of the client rendering area.

**Review by madebr — APPROVED** [review]
Also passing here
```
INFO:  03/18/23 21:33:40: Assert Summary: Total=104 Passed=104 Failed=0
INFO:  03/18/23 21:33:40: Total Test runtime: 1.5 sec
INFO:  03/18/23 21:33:40: >>> Test 'video_setWindowCenteredOnDisplay': Passed
INFO:  03/18/23 21:33:40: Total Suite runtime: 1.5 sec
```

**Comment by 1bsyl** [comment]
I've updated the PR, before I thought the position was top-left of decoration bar. whereas it's top left of the effective rendering area (so a smaller diff).

I also checked the other suite like video_getSetWindowPosition().
WM doesn't allow to set a window the position to (0, 0) for instance because of the desktop menu/status bar.
so it puts it to (136, 54) for instance. 
Also same for size and video_getSetWindowSize.

a solution would be to replace XMoveWindow() and X11_XResizeWindow()
by a request _NET_MOVERESIZE_WINDOW ... but that doesn't seem to work
and maybe we prefer, the current behavior: 
if app sets its position to 0,0,  let WM set it at (136, 54) so it doesn't hide the status bar ?!

File: src/video/x11/SDL_x11window.c (hunk 1 above)
Comment (1bsyl) [reply src/video/x11/SDL_x11window.c:1534]: ok, my mistake, I've reverted that

**Comment by slouken** [comment]
Yes, the expected behavior is that SDL_SetWindowPosition() and SDL_SetWindowSize() will make requests of the window manager and set the internal position and size, and then we get events signaling what actually happened, and they get passed on to the application if they're not what we expect.

**Comment by 1bsyl** [comment]
updated, now SetSize/Position pass for X11
in the tests, we now checks for MOVE/RESIZED event as fallback

**Comment by 1bsyl** [comment]
And previous commit is fixing the grab test by requesting input focus.  + Update window flags to have input focus. not sure if this is the right thing ...

**Comment by 1bsyl** [comment]
on x11, 
it now only fails with --filter render_testLogicalSize
when comparing to reference  image:  CompareSurfaces0001_Reference.bmp 2 and 3

**Comment by 1bsyl** [comment]
updated: 
better fix for video_getSetWindowGrab(): test needs to raise the window, and wait for focus gained.

**Comment by 1bsyl** [comment]
merging thoses fixes!

**Comment by 1bsyl** [comment]
@sezero @slouken 
I wonder whether the two x11 video backends should be cherry-picked for SDL2 ? 
https://github.com/libsdl-org/SDL/pull/7502/commits/77df804d1604f3273e9bbb852300bc912b2f5831
https://github.com/libsdl-org/SDL/pull/7502/commits/93802db109c846fdb06c6b727c179831bb30b83f

**Comment by slouken** [comment]
Yes, that seems fine

**Comment by 1bsyl** [comment]
ok done : https://github.com/libsdl-org/SDL/commit/2dddaa7dc9d354595526faca537232579a9e9265

**Pull request merged**